File: Easy/2026-08-18-3471-Find-the-Largest-Almost-Missing-Integer/solution.java
import java.util.HashMap;
import java.util.Map;

public class Solution {
    public int findLargestAlmostMissingInteger(int[] nums, int k) {
        Map<Integer, Integer> countMap = new HashMap<>();
        int n = nums.length;

        for (int i = 0; i <= n - k; i++) {
            for (int j = i; j < i + k; j++) {
                countMap.put(nums[j], countMap.getOrDefault(nums[j], 0) + 1);
            }
        }

        int largestAlmostMissing = -1;
        for (Map.Entry<Integer, Integer> entry : countMap.entrySet()) {
            if (entry.getValue() == 1) {
                largestAlmostMissing = Math.max(largestAlmostMissing, entry.getKey());
            }
        }

        return largestAlmostMissing;
    }
}

File: Easy/2026-08-18-3471-Find-the-Largest-Almost-Missing-Integer/solution.py
class Solution:
    def findLargestAlmostMissingInteger(self, nums: List[int], k: int) -> int:
        from collections import defaultdict
        
        count = defaultdict(int)
        n = len(nums)
        
        for i in range(n - k + 1):
            unique_elements = set(nums[i:i + k])
            for num in unique_elements:
                count[num] += 1
        
        largest_almost_missing = -1
        for num in count:
            if count[num] == 1:
                largest_almost_missing = max(largest_almost_missing, num)
        
        return largest_almost_missing